shared/src/main/scala/io/estatico/newtype/NewType.scala

```diff
@@ -7,6 +7,10 @@ trait NewType extends BaseNewType { self =>
 }
 
 object NewType {
+  class NewTypeUnapplyOps[A](val a: A) extends AnyVal {
+    def isEmpty: Boolean = false
+    def get: A = a
+  }
```

Comment:
Should probably be `final class`

Reply:
I also wonder if you could get away with using `val get` in the constructor, e.g.

```scala
final class NewTypeUnapplyOps[A](val get: A) extends AnyVal {
  def isEmpty: Boolean = false
}
```